README.md

```diff
@@ -1,2 +1,65 @@
 # action-cli-runscript
 GitHub action to run a script using the Office 365 CLI
+
+![Office 365 CLI Runscript](./images/cli-runscript.png)
+
+This GitHub Action (created using typescript) uses [Office 365 CLI](https://pnp.github.io/office365-cli/), to run a line of script supplied to it or run code in a script file supplied to it.
+
+## Usage
+### Pre-requisites
+Create a workflow `.yml` file in `.github/workflows` directory of your repo. An [example workflow](#example-workflow---office-365-cli-runscript) is available below. For more information, reference the GitHub Help Documentation for [Creating a workflow file](https://help.github.com/en/articles/configuring-a-workflow#creating-a-workflow-file).
+
+### Note
+This action is dependant on `action-cli-login`. So in the workflow we need to run  `action-cli-login` before using this action.
+
+#### Optional requirement
+Since `action-cli-login` requires user name and password which are sensitive pieces of information, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add 2 new secrets:
+- `adminUsername` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
+- `adminPassword` - store the password of that user in this.
+These secrets are encrypted and can only be used by GitHub actions.
+
+### Inputs
+- `O365_CLI_SCRIPT` : The script to run
+- `O365_CLI_SCRIPT_PATH` : Relative path of the script in your repo.
+- `IS_POWERSHELL` : `true|false` Used only with O365_CLI_SCRIPT. Default is true. If false the assumption is the script will be shell script i.e. .sh.
```

Comment:
@anoopt , @garrytrinder , @waldekmastykarz  would you all agree to default to PowerShell. Please see my comments below before answer.

#### It seems running `bash` is significantly faster than PowerShell.  See my tests: 

To reproduce the speed difference I am running the script action with `o365 spo tenant appcatalogurl get`. See the full workflow [here](https://github.com/VelinGeorgiev/spfx-react-router/blob/master/.github/workflows/deploy.yml).

|OS  | Bash   |PowerSell    | 
|---|---|---|
| Linux   |[2-3 sec](https://github.com/VelinGeorgiev/spfx-react-router/commit/a893b8fefaf89dc7b7f5af105795461f427b5197/checks?check_suite_id=400037730)   | [10-18 sec](https://github.com/VelinGeorgiev/spfx-react-router/commit/712cbe80776567522b02c5fe98e604fcf493cc1e/checks?check_suite_id=400041337)  |
| Mac   |[2-3 sec](https://github.com/VelinGeorgiev/spfx-react-router/commit/517b215ff2d906ee5ccd8d5fa9ef543cb48e5d6c/checks?check_suite_id=400066785)   | [7 sec](https://github.com/VelinGeorgiev/spfx-react-router/commit/353213257b4ff051206ca4a5f02d79106bf9e4de/checks?check_suite_id=400062186)   |
| Windows  | [Error that we would have to fix](https://github.com/VelinGeorgiev/spfx-react-router/commit/de5f6bc0c07b6d98df853bc3430b7bda69b57531/checks?check_suite_id=400048873)   |[14 sec](https://github.com/VelinGeorgiev/spfx-react-router/commit/06badc50befe524215598f159dc6efb9fba4dbab/checks?check_suite_id=400056123)   | 

#### In case of self-hosted by the customer VM runners, we might have to add more info in the README regardless of which one is default.

If the users use self hosted VM, they might not have `powershell` nor `bash` installed. It might be nice to have a section in the README file that explains that the user must install `powershell` or `bash`  if they plan to use self-hosted VM runners.

Reply:
@VelinGeorgiev - Thank you for providing the results. Those numbers clearly make sense to use bash as default. I will update the code accordingly.